lib/redis_memo/memoize_query.rb

```diff
@@ -9,10 +9,35 @@ module RedisMemo::MemoizeQuery
   require_relative 'memoize_query/invalidation'
   require_relative 'memoize_query/model_callback'
 
-  # Only editable columns will be used to create memos that are invalidatable
-  # after each record save
+  # Core entry method for using RedisMemo to cache SQL select queries on the given
+  # column names.
+  #
+  #   class User < ApplicationRecord
+  #     extend RedisMemo::MemoizeQuery
+  #     memoize_table_column :id
+  #     memoize_table_column :first_name
+  #     memoize_table_column :first_name, last_name
+  #   end
+  #
+  # On the User model, queries such as
+  #   - record.user
+  #   - User.find(user_id)
+  #   - User.where(id: user_id).first
+  #   - User.where(first_name: first_name).first
+  #   - User.where(first_name: first_name, last_name: last_name).first
+  #   - User.find_by_first_name(first_name)
+  # will first check the Redis cache for the data before hitting the SQL database;
+  # the cache results are invalidated automatically when user records are changed.
+  #
+  # Note that +memoize_table_column :first_name, last_name+ specifies that only AND queries
+  # that contain both columns will be memozed. The query +User.where(last_name: last_name)+
+  # will NOT be memoized with the given configuration.
+  #
+  # @param raw_columns [Array] A list of columns to memoize.
+  # @param ediable [Boolean] Specify if the column is editable. Only editable columns
```

Comment:
typo
```suggestion
  # @param editable [Boolean] Specify if the column is editable. Only editable columns
```
<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia3.giphy.com%2Fmedia%2F12uXi1GXBibALC%2Fgiphy.gif&token=food" data-gitmeme-token="food"><img src="https://media3.giphy.com/media/12uXi1GXBibALC/giphy.gif" title="Created by gitme.me with /food"/></a>